### PR TITLE
Issue 4859: Interactive Demo Tool

### LINKED ISSUE #4859: Interactive Demo Tool
**Problem description**
Create an interactive demo tool that uses the Pravega Client to execute various Scope/Stream/KVT operations on Pravega.

**Problem location**
Interactive Demo.

**Suggestions for an improvement**
Make the tool.

### PR DESCRIPTION
**Pull request by andreipaduroiu**
**Change log description**  
- Initial implementation of Pravega's IDT (Interactive Demo Tool). Supports Scopes, Streams and KVTs.
- Simple Chat Service using Pravega Streams, KVTs and Stream Cuts.

**Purpose of the change**  
Fixes #4859.

**What the code does**  
The IDT uses the Pravega Client to do the following operations:
- Scopes: create and delete.
- Streams: create, delete, list, append, read.
- KVTs: create, delete, put, get, remove, list keys/entries.

**How to verify it**  
1. Run standalone: `./gradlew startStandalone`
2. Run the tool. `./gradlew startInteractiveDemo -q --console=plain`
    - `-q` and `--console=plain` are required so that Gradle will get out of the way of console IO.

(Wiki page will be built [here](https://github.com/pravega/pravega/wiki/Interactive-Demo-Tool) probably after this PR is merged in).

### CODE REVIEW

**Comment by eolivelli** [comment]
Why not moving this tool to "pravega-cli" project ?
This way we can keep a cleaner codebase for this core repository

**Comment by RaulGracia** [comment]
@eolivelli that's a good suggestion, and I tend to agree with you at least up to some extent. This PR contains actually 2 demos: one to show the raw KVT capabilities and semantics, and another that builds a nice chat on top of that (and using some other Pravega primitives). I already mentioned to @andreipaduroiu if it would be possible to move at least the chat demo code to Pravega Samples, as this is the repo that we officially suggest as a starting point for new users wanting to play with Pravega. The goal would be to maximize the visibility of these demos, as they are a very nice starting point for someone willing to understand the new KVT feature.

File: test/integration/src/main/java/io/pravega/test/integration/demo/interactive/BackgroundConsoleListener.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.util.Scanner;
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.val;
+
+/**
+ * Acquires console input on a background thread and updates its state ({@link #isTriggered()} when a desired token
+ * has been typed in.
+ */
+class BackgroundConsoleListener implements AutoCloseable {
```
Comment (eolivelli) [inline test/integration/src/main/java/io/pravega/test/integration/demo/interactive/BackgroundConsoleListener.java:20]: What about using JLine  ?

It  provides a few useful tools, like auto completion and history,

I have been playing with Pravega CLI during the last days and not having autocomplete and history is really awkward 

IMO for this very first release of the tool we can live without JLine.

But we should take a little bit of UX into consideration, especially for this kind of demo tools and are the first way of approaching the product by people who are deciding to adopt it.  


cc @derekm

File: test/integration/src/main/java/io/pravega/test/integration/demo/interactive/BackgroundConsoleListener.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    private final AtomicBoolean triggered = new AtomicBoolean(false);
+    private final String token;
+
+    BackgroundConsoleListener() {
+        this("q");
```
Comment (tkaitchuck) [inline test/integration/src/main/java/io/pravega/test/integration/demo/interactive/BackgroundConsoleListener.java:25]: Why q?

**Comment by andreipaduroiu** [comment]
My apologies for closing this. There were too many problems retargeting this PR onto master so I opened a new one to get this in. #4928

**Pull request closed without merging**